python/sidecar/references/schemas.py

```diff
@@ -76,8 +89,30 @@ class ReferenceStatus(RefStudioModel):
     status: IngestStatus
 
 
+class IngestRequestType(StrEnum):
+    UPLOADS_DIRECTORY = "uploads"
+    PDF = "pdf"
+    GENERIC = "generic"
+
+
 class IngestRequest(RefStudioModel):
-    pdf_directory: str
+    type: IngestRequestType
+
+
+class IngestUploadsRequest(IngestRequest):
+    type: IngestRequestType = IngestRequestType.UPLOADS_DIRECTORY
+    pass
+
+
+class IngestPdfUrlRequest(IngestRequest):
+    type: IngestRequestType = IngestRequestType.PDF
+    url: str
+    metadata: ReferenceCreate
+
+
+class IngestGenericRequest(IngestRequest):
+    type: IngestRequestType = IngestRequestType.GENERIC
+    metadata: ReferenceCreate
```

Comment:
Not quite sure how y'all would prefer to handle this on the frontend.

Since we are now allowing users to create references w/o PDFs, it might make sense for us to drop `IngestPdfUrlRequest` and instead just use `IngestGenericRequest` and add an optional `url` parameter. So two potential request types, rather than three.

Do you have thoughts @cguedes @jvoytek?

Reply:
So long as available PDFs from S2 are still ingested 2 types is fine with me.

Reply:
I don't mind 2 types. But would call it then `IngestMetadataRequest`, given that the PDF is now optional and generic is too abstract.